### PR TITLE
Allow users to query for Handlers from the GraphQL endpoint

### PR DESCRIPTION
**Pull request by wesleycyu**
## What is this change?
This PR adds the ability to query for Handlers from the GraphQL schema to support sensu/web#89. There are two ways users can retrieve handlers. 
1. Users can retrieve a single `Handler` given a namespace and a name from the `handler` field on the root `query` type,
2. Or, users can retrieve a _collection_ of handers (`HandlerCollection`) from the `handlers` field on the `namespace` type.

##### Example Single `handler` query
Screen Shot 2019-05-13 at 11 49 31 AM

##### Example Collection `handlers` query
Screen Shot 2019-05-13 at 11 48 55 AM

## Why is this change necessary?
This change supports a future update in the in the [sensu/web](https://github.com/sensu/web) repo (sensu/web#89), Sensu users want the ability to view a list of handers and see the details of a particular selected handler. This change allows the Web UI to query for handlers in order to display this data to the user. 

## Does your change need a Changelog entry?
Yes, 5aae0c4

## How did you verify this change?
To manually verify this change you can run a `sensu-backend` and query the GraphQL endpoint via GraphiQL. 
1. Run a `sensu-backend` locally
2. Create a number of handlers. I did this using `sensuctl -create --file example_handlers.json` ([example_handlers.json](https://gist.github.com/wesleycyu/dd3dacaff7b993f6926717ff5fe00fa4))
3. Run the [sensu/web](https://github.com/sensu/web) application locally with the apollo dev tools Chrome extension installed. (Don't forge to update the `graphQLSchemaRef` in `package.json` to the latest commit on this PR `c29c060`)
4. Run queries to ensure that handlers are being returned correctly.

*example query:*
```
{
  handler(namespace: "default", name: "slack") {
    id
    name
    namespace
    command
    envVars
  }
  namespace(name: "default") {
    handlers {
      nodes {
        id
        name
        command
        handlers {
          id
        }
        socket {
          host
          port
        }
      }
    }
  }
}
```

### CODE REVIEW

**Comment by wesleycyu** [comment]
@jamesdphillips  -- Do GraphQL schema changes require tests in the [sensu/sensu-go-qa-crucible](https://github.com/sensu/sensu-go-qa-crucible) repo? I didn't see tests in there for other GraphQL stuff, so I didn't create a PR to test my changes from this PR.

**Comment by jamesdphillips** [comment]
> Do GraphQL schema changes require tests in the sensu/sensu-go-qa-crucible repo? I didn't see tests in there for other GraphQL stuff, so I didn't create a PR to test my changes from this PR.

Not at this time.

File: backend/apid/graphql/namespace.go [hunk 1]
```diff
@@ -103,6 +103,48 @@ func (r *namespaceImpl) Checks(p schema.NamespaceChecksFieldResolverParams) (int
...
+		records,
+		p.Args.OrderBy == schema.HandlerListOrders.NAME,
+	))
+
+	// paginate
+	l, h := clampSlice(p.Args.Offset, p.Args.Offset+p.Args.Limit, len(records))
+	res.Nodes = records[l:h]
+	res.PageInfo.totalCount = len(records)
+	return res, nil
+}
```
Comment (jamesdphillips) [inline backend/apid/graphql/namespace.go]: I know this is a bit of 'do as I say, not as I do' but would you mind adding a test for this field resolver? There should be a test for the silences field resolver that you could use as a template.

**Comment by portertech** [comment]
730147c4-7560-11e4-9375-fefd2f67563b

**Comment by wesleycyu** [comment]
Thanks for the review @jamesdphillips @dabria @portertech!

I've added a test for the `handlers` field on the `namespace` type (7ebd44b) :)

**Pull request merged**